zerver/lib/alert_words.py

```diff
@@ -1,17 +1,18 @@
-from django.db.models import Q
-from zerver.models import UserProfile, Realm
+from zerver.models import UserProfile, Realm, AlertWords
 from zerver.lib.cache import cache_with_key, realm_alert_words_cache_key, \
     realm_alert_words_automaton_cache_key
-import ujson
 import ahocorasick
 from typing import Dict, Iterable, List
 
 @cache_with_key(realm_alert_words_cache_key, timeout=3600*24)
 def alert_words_in_realm(realm: Realm) -> Dict[int, List[str]]:
     users_query = UserProfile.objects.filter(realm=realm, is_active=True)
-    alert_word_data = users_query.filter(~Q(alert_words=ujson.dumps([]))).values('id', 'alert_words')
-    all_user_words = dict((elt['id'], ujson.loads(elt['alert_words'])) for elt in alert_word_data)
-    user_ids_with_words = dict((user_id, w) for (user_id, w) in all_user_words.items() if len(w))
+    user_ids_with_words = dict()
+    for user_profile in users_query:
+        alert_words_objects = AlertWords.objects.filter(user_profile=user_profile)
+        if len(alert_words_objects):
+            alert_words = [alert_word.word for alert_word in alert_words_objects]
+            user_ids_with_words[user_profile.id] = alert_words
```

Comment:
This loop structure is really expensive.  I think we should just denormalize a `realm` column in the new `AlertWord` table, and have an index on that column, so this can be `AlertWords.objects.filter(realm=realm).values("user_profile_id", "alert_words")`.